### DISCUSSION TITLE
DefaultContentManager does not support collection

### DISCUSSION THREAD

**Discussion by duxonline** [OrchardCMS/OrchardCore#14813] [General]
I was trying to save content items from my custom module in a different table instead of `dbo.Document`. For example, I thought I could save my content items in `dbo.Forum_Document`. It doesn't seem I can achieve this using DefaultContentManager though. 

In DefaultContentManager, when we call `Session.Save`, we ignore `collection` parameter:
 `_session.Save(contentItem, checkConcurrency: true)`

This is the signature of Session.Save:
`public void Save(object entity, bool checkConcurrency = false, string collection = null)`

**Comment by MikeAlhayek** [comment]
The content definition manager is intended for storing ContentItems in the default store. If you want to use a custom store, you should use _session.Save instead.Or implement your own IContentItemManager.

**Reply by duxonline** [reply]
Thank you for the reply. It's good to know that it's not a crazy idea to implement my own IContentItemManager.

**Reply by MikeAlhayek** [reply]
I never said it wasn't :) 

Why do you need to use a separate collection?

**Reply by duxonline** [reply]
I am trying to use a different collection to avoid naming conflicts.

**Reply by MikeAlhayek** [reply]
I am not sure this is a recommended approach. You are probably no considering the amount of work you'll need. Not sure you need the complexity. You'll need to add collection support for your content types. you'll also need the UI to support multiple collections. It  may be much simpler to rename some of your contents types so there is no naming conflict instead.

**Reply by duxonline** [reply]
Thank you for the information. I will keep it simple for now.

**Reply by hiteshnsharma** [reply]
@MikeAlhayek I have similar requirement. I am working on an application which will majorly have two type of content with each having a lot of fields, by default they will be stored in document table in json format. Since I am expecting a large amount of data will end up in document table (millions of records), I would prefer having separate table for these content types, This will also make data management easy (I assume) where these content types will have their own tables with columns representing their fields.

Do you think this a valid use case to have separate document table for these content types?

@duxonline Did you find any solution for your use case?

**Reply by MikeAlhayek** [reply]
@hiteshnsharma if you think your content type need a separate document table, sure why not. But this means that you'll need to create your own UI for this content type. Also, you'll need to create a new service similar to the DefaultContentManager where it would store the documents into a separate collection instead of the default one.

Alternatively, you could try to come up with a way to make the default content manager configurable where you could change the default collection for your needs.

**Reply by duxonline** [reply]
@hiteshnsharma I took Mike's advice and store all content items in default document table.

**Comment by ns8482e** [comment]
There is built-in yessql support to have different document store table if you wish and it doesn't require you to implement your own IContentManager. However remember that then all indexing will be separate than default content item document store.

Take an example code of Open ID module, it uses separate document store

**Reply by duxonline** [reply]
Thank you @ns8482e
